//
// Copyright 2019 Asylo authors
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//     http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.
//

syntax = "proto2";

package asylo;

import "asylo/enclave.proto";

// Configuration for an SGX enclave loader.
message SgxLoadConfig {
  extend EnclaveLoaderConfig {
    optional SgxLoadConfig sgx_load_config = 263000381;
  }

  // Path to the enclave binary (.so) file to load.
  optional string enclave_path = 1;

  // True, if the enclave is to be loaded in debug mode.
  optional bool debug = 2;

  // Configuration passed to an enclave during initialization.
  optional EnclaveConfig config = 3;

  message ForkConfig {
    // Start address to load enclave.
    optional uint64 base_address = 1 [default = 0];

    // Size of the enclave in memory.
    optional uint64 enclave_size = 2 [default = 0];
  }

  // Configuration passed by the implementation of fork to load a child enclave
  // in the same virtual space as the parent.
  optional ForkConfig fork_config = 4;
}
